### DISCUSSION TITLE
DeckGL incompatible with Geocoder?

### DISCUSSION THREAD

**Discussion by mmz15** [visgl/deck.gl#6001] [Q&A] [answered]
I'm trying to add geocoding to my DeckGL/Mapbox project. The result looks fine on the surface:

<img width="957" alt="Screen Shot 2021-07-21 at 20 16" src="https://user-images.githubusercontent.com/69217396/126575227-6a943b8b-94e1-47a1-92ba-1d81a7d48927.png">

The problem, though, is that I can't click on the search box. It's as though it's an extension of the map. Here's the code that generated the above:
```
      <DeckGL
        {...deckGLProps}
        ref={deckRef}
        style={{ position: "absolute", zIndex: "-9999" }}
      >
        <StaticMap
          ref={mapRef}
          {...staticMapProps}
          style={{ position: "absolute", zIndex: "-9999" }}
          mapStyle={link-to-style}
        >
        <Geocoder
          mapRef={mapRef}
          onViewportChange={handleGeocoderViewportChange}
          mapboxApiAccessToken={process.env.REACT_APP_MAPBOX_TOKEN}
          position="top-left"
          minLength={1}
          trackProximity={true}
          countries={"us"}
          reverseGeocode={true}
          enableEventLogging={false}
        />
        </StaticMap>
      </DeckGL>
```

The `Geocoder` component is from the `react-map-gl-geocoder` package. 

I suspect that this is a component-ordering issue. When analyzing the HTML, the  Geocoder `input` is nested in the `mapboxgl-control-container`:
<img width="725" alt="Screen Shot 2021-07-21 at 20 20" src="https://user-images.githubusercontent.com/69217396/126575444-a47e3edf-704b-4fc1-ac6c-0f976358c0dc.png">

Wrapping the `Geocoder` in a `div` with a high `zIndex` doesn't help. Interestingly, the `div` doesn't even wrap the component:
<img width="713" alt="Screen Shot 2021-07-21 at 20 24" src="https://user-images.githubusercontent.com/69217396/126575704-089d3041-9cd3-4faf-ae48-c75ea71c58e0.png">

So my question is straightforward: Is there a way to set a `zIndex` on the `Geocoder` component? Is there another way to organize the components so that the search bar is clickable?

**Comment by Pessimistress** [comment, marked as answer]
See https://github.com/visgl/deck.gl/discussions/5834

**Reply by mmz15** [reply]
Great context, thanks as always @Pessimistress 

Also, [this](https://stackoverflow.com/questions/68474118/mapbox-react-unable-to-input-values-into-search-box/68485777#68485777) also helped for anyone passing by
